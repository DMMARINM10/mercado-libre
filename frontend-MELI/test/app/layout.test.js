import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { useRouter, usePathname, useSearchParams } from 'next/navigation';
import RootLayout from '@/app/layout';

const child = <div data-testid="test-child">Child Component</div>
const inputPlaceholder = 'Nunca dejes de buscar'

// Mock the Next.js router
jest.mock('next/navigation', () => ({
    useRouter: jest.fn(),
    usePathname: jest.fn(),
    useSearchParams: jest.fn(),
}));
  
const mockRouter = {
  push: jest.fn(),
};
  
beforeEach(() => {
  useRouter.mockReturnValue(mockRouter);
  usePathname.mockReturnValue('/items');
  useSearchParams.mockReturnValue({
    get: jest.fn().mockReturnValue('test search'),
  });
});

describe('Main Layout component', () => {
  it('should render SearchBar and children correctly', async() => {
    render(
      <RootLayout>
        {child}
      </RootLayout>
    );

    //Testing SearchBar is in the document
    const input = screen.getByPlaceholderText(inputPlaceholder);
    expect(input).toBeInTheDocument();

    const childComponent = screen.getByTestId('test-child');
    expect(childComponent).toBeInTheDocument();
  });

  it('should match the snapshot', () => {
    const { asFragment } = render(<RootLayout>{child}</RootLayout>);
    expect(asFragment()).toMatchSnapshot();
  });
});
