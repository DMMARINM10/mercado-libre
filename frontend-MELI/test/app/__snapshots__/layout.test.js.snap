// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Layout component should match the snapshot 1`] = `
<DocumentFragment>
  <header
    class="search-bar-container"
  >
    <img
      alt="Mercado Libre Logo"
      class="logo"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="50"
      src="/_next/image?url=%2Fimg.jpg&w=128&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=64&q=75 1x, /_next/image?url=%2Fimg.jpg&w=128&q=75 2x"
      style="color: transparent;"
      width="50"
    />
    <fieldset>
      <label
        for="input-search-bar"
        hidden=""
      />
      <input
        id="input-search-bar"
        placeholder="Nunca dejes de buscar"
        type="text"
        value="test search"
      />
      <button>
        <img
          alt="Search icon"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="15"
          src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
          style="color: transparent;"
          width="15"
        />
      </button>
    </fieldset>
  </header>
  <main
    class="content"
  >
    <div
      data-testid="test-child"
    >
      Child Component
    </div>
  </main>
</DocumentFragment>
`;
